hack/tests/upgrade-e2e.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -ex
+# upgrade-e2e script is intended to run in CI environmet and will not run localy
```

Comment:
this is awfully similar to the above file - how can it be done differently? is a single bash file using functions any better?

Reply:
We have 2 files `upgrade-e2e` and `create-e2e` and they have a lot of common code. I already unified `upgrade-e2e` to be able to run locally. I dont want to merge them as it will make them even more complex :/ keep it simple, stupid.